### PR TITLE
Journalist can categorize Articles

### PR DESCRIPTION
**Pull request by mantanquzzmo**
[PT-Link](https://www.pivotaltracker.com/story/show/170675601)
As a journalist
In order to distinguish my articles to a relevant topic
I would like to be able to associate them with different categories

- Adds category to Article model
- Adds category to serializers

### CODE REVIEW

**Comment by EevanR** [comment]
lgtm

File: app/controllers/api/v1/articles_controller.rb [hunk 1]
```diff
@@ -2,8 +2,13 @@ class Api::V1::ArticlesController < ApplicationController
   rescue_from ActiveRecord::RecordNotFound, with: :article_not_found
 
   def index
-    articles = Article.paginate(page: params[:page], per_page: 4)
-    render json: articles, each_serializer: Articles::IndexSerializer, meta: meta_attributes(articles)
+    if params[:category]
+      articles = Article.where(:category => params[:category]).paginate(page: params[:category], per_page: 4)
```
Comment (sundqvistjohan) [inline app/controllers/api/v1/articles_controller.rb]: use `Article.where(category: params[:category])...` instead to be consistent

File: app/controllers/api/v1/articles_controller.rb [hunk 2]
```diff
@@ -2,8 +2,13 @@ class Api::V1::ArticlesController < ApplicationController
...
+      render json: articles, each_serializer: Articles::IndexSerializer, meta: meta_attributes(articles)
+    else
+      articles = Article.paginate(page: params[:page], per_page: 4)
```
Comment (sundqvistjohan) [inline app/controllers/api/v1/articles_controller.rb:8]: Do we want to have uncategorized articles?

File: app/controllers/api/v1/articles_controller.rb (hunk 1 above)
Comment (mantanquzzmo) [reply app/controllers/api/v1/articles_controller.rb]: ok

File: app/controllers/api/v1/articles_controller.rb (hunk 2 above)
Comment (mantanquzzmo) [reply app/controllers/api/v1/articles_controller.rb:8]: we have a default Misc, and for now at least we want the herald front page so have a nice mix.

**Comment by EevanR** [comment]
LGTM

**Review by sundqvistjohan — APPROVED** [review]

File: app/controllers/api/v1/articles_controller.rb [hunk 3]
```diff
@@ -2,8 +2,13 @@ class Api::V1::ArticlesController < ApplicationController
   rescue_from ActiveRecord::RecordNotFound, with: :article_not_found
 
   def index
-    articles = Article.paginate(page: params[:page], per_page: 4)
-    render json: articles, each_serializer: Articles::IndexSerializer, meta: meta_attributes(articles)
+    if params[:category]
+      articles = Article.where(category: params[:category]).paginate(page: params[:category], per_page: 4)
+      render json: articles, each_serializer: Articles::IndexSerializer, meta: meta_attributes(articles)
```
Comment (oliverochman) [inline app/controllers/api/v1/articles_controller.rb]: you can out this line after the conditional

code repetition on line 7 and 10

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/articles_controller.rb (hunk 3 above)
Comment (mantanquzzmo) [reply app/controllers/api/v1/articles_controller.rb]: fixed

**Review by oliverochman — APPROVED** [review]

**Pull request merged**